gradio/blocks.py

```diff
@@ -1110,26 +1098,38 @@ def reverse(text):
                     )
         utils.launch_counter()
 
+        self.is_colab = utils.colab_check()
+        self.share = (
+            share
+            if share is not None
+            else True
+            if self.is_colab and self.enable_queue
+            else False
+        )
+
         # If running in a colab or not able to access localhost,
         # a shareable link must be created.
-        is_colab = utils.colab_check()
-        if is_colab or (_frontend and not networking.url_ok(self.local_url)):
+
+        if self.is_colab and not quiet:
+            if debug:
+                print(strings.en["COLAB_DEBUG_TRUE"])
+            else:
+                print(strings.en["COLAB_DEBUG_FALSE"])
+
+        if (self.is_colab and not self.share) or (
```

Comment:
I believe this should be:
```suggestion
        if (self.is_colab and and self.enable_queue and not self.share) or (
```

The conditions in this line are hard to parse. Consider refactoring. Also, can you add tests to make sure that all of the combinations of share/queue/colab are correctly handled?